### DISCUSSION TITLE
Can I use a URL as the source of a page?

### DISCUSSION THREAD

**Discussion by ydkmlt84** [squidfunk/mkdocs-material#7861] [Q&A] [answered]
Can I use the actual raw file from a GitHub repo as the source of my page? Instead of a link to that URL?

For example, I want to add a page to my docs, that the actual markdown is located inside a file of a repo.

`    - Installation: https://github.com/repo/raw/Installation.md`

Right now, it just creates a link to that file. Thank you again for any help/information.

**Comment by squidfunk** [comment]
That's currently not supported, but one might be able to write an MkDocs plugin for that! Note that this is a limitation of MkDocs, not Material for MkDocs, so it might also be beneficial to ask on their discussion board.

**Reply by kamilkrzyskow** [reply]
If the syntax in the `nav` isn't crucial, then there is the [Snippets](https://facelessuser.github.io/pymdown-extensions/extensions/snippets/) extension, which allows to include a file from the web / outside the `docs` dir. The file can also be sliced to avoid double headings.
The issues could include, having to create a `Installation.md` file in the `docs` and inside of it, include the snippet, or perhaps the file is written with CommonMark implementation for GitHub and some parts would break in Python Markdown.

Reference:
- https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown-extensions/#snippets

**Reply by ydkmlt84** [reply, marked as answer]
Snippets worked great! 

I added this to my mkdocs.yml
```
markdown extensions:
  - pymdownx.snippets:
      url_download: true
```
Then made a file Changelog.md and added it to my nav setup like my other pages.

This is what is inside my Changelog.md file
```
---
title: Changelog
description: Github Changelog file for Maintainerr.
---

--8<-- "URL of page where the file is"
```

This is the outcome. Thanks again! I knew snippets was a thing, but didn't know it did this or how to use it.

![image](https://github.com/user-attachments/assets/0d2271a5-25c2-42ac-ad0c-b2aba5826042)
